### PR TITLE
391 Fixing Tests and Updating Images

### LINKED ISSUE #391: Error running `make test` with conda env
After running the Dockerfile for [gp-benchmarking](https://github.com/matsengrp/gp-benchmark-1-environment). I noticed that the `TEST_CASE("RootedSBNInstance: gradients") {` -> [Here](https://github.com/phylovi/bito/blob/098c1f0f278517f6b000b0d88e5cfe001948cef0/src/rooted_sbn_instance.hpp#L260)
is failing. I'm not sure if this is a known error, but in either case the `make test` command should probably be run in the [bito container](https://github.com/phylovi/bito/blob/main/Dockerfile) and/or a [workflow](https://github.com/phylovi/bito/blob/main/.github/workflows/build-with-cmake.yml) so errors like this can be caught right away.

### PR DESCRIPTION
**Pull request by tanviganapathy**
## Description

#391 error fixed by using an ordered map for PCSPCounter which guarantees the PCSP index order (#402). 

Summary of changes made to reflect new PCSP indexing:
- Tests are updated.
- Updated images are linked as comments in issue #391. Old images are also linked for reference.
- Links in code point to new images.
- Images in `doc/svg` are updated.
- Minor changes to reflect new naming conventions.

Closes #391

## Checklist:

* [x] Comments are up to date, document intent, and there are no commented-out code blocks
* [x] TODOs have been eliminated from the code
* [x] The corresponding issue number (e.g. `#391`) has been searched for in the code to find relevant notes

### CODE REVIEW

File: src/gp_doctest.cpp [hunk 1]
```diff
@@ -573,10 +565,10 @@ TEST_CASE("GPInstance: GenerateCompleteRootedTreeCollection") {
   inst.GetEngine()->SetBranchLengths(branch_lengths);
   // Because the branch lengths contain the GPCSP index, we can check that the indices
   // correspond to what we see in the GPCSP DAG in
-  // https://github.com/phylovi/bito/issues/349#issuecomment-897233859
+  // https://github.com/phylovi/bito/issues/391#issuecomment-1168048090
   CHECK_EQ(inst.GenerateCompleteRootedTreeCollection().Newick(),
-           "((0:7,1:9):3,(2:11,(3:12,4:13):5):2):0;\n"
-           "(1:10,(0:8,(2:11,(3:12,4:13):5):6):4):0;\n");
+           "((0:7,1:9):3,(2:11,(3:12,4:13):2):6):0;\n"
+           "(1:10,(0:8,(2:11,(3:12,4:13):2):5):4):0;\n");
```
Comment (davidrich27) [inline src/gp_doctest.cpp:571]: Just curious, why were these branch lengths changed?

**Comment by matsen** [comment]
@davidrich27 did the left/right changes make sense to you? Either way, I'd like to know why they came about.

**Comment by matsen** [comment]
Oops! Didn't mean to close.

**Comment by davidrich27** [comment]
I'll discuss it with Tanvi.

File: src/gp_doctest.cpp (hunk 1 above)
Comment (tanviganapathy) [reply src/gp_doctest.cpp:571]: I don't think the branch lengths actually changed. Those numbers refer to the edge idxs above that node which were changed when PCSPCounter became an ordered map.

**Comment by davidrich27** [comment]
This was due to some mislabeling on my part when I switched over to enums.  It said `SubsplitClade::LeftChild`, but it should have been `SubsplitClade::RightChild` from the start.  The logic hasn't changed.

File: src/gp_doctest.cpp [hunk 2]
```diff
@@ -535,18 +529,18 @@ TEST_CASE("GPInstance: inverted GPCSP probabilities") {
...
+      1.,                            // 3
+      1.,                            // 4
+      2. / 3.,                       // 5
       // We have the 0.5 coming from node 12, but that's split evenly between the two
       // descendants, so we have 0.25 from each. Thus even weights.
       0.5,      // 6
-      1.,       // 7
-      1.,       // 8
+      0.5,      // 7
+      0.5,      // 8
```
Comment (matsen) [inline src/gp_doctest.cpp:539]: I don't follow this one: the two nodes are 0.25 and 0.5. 

I'm sure you are right though!

**Review by matsen — APPROVED** [review]

**Pull request merged**